packages/luma/src/luma-image-model.ts

```diff
@@ -0,0 +1,274 @@
+
+    if (seed != null) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'seed',
+        details: 'This model does not support the `seed` option.',
+      });
+    }
+
+    if (size != null) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'size',
+        details:
+          'This model does not support the `size` option. Use `aspectRatio` instead.',
+      });
+    }
+
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+    const fullHeaders = combineHeaders(this.config.headers(), headers);
+    const { value: generationResponse, responseHeaders } = await postJsonToApi({
+      url: this.getLumaGenerationsUrl(),
+      headers: fullHeaders,
+      body: {
+        prompt,
+        ...(aspectRatio ? { aspect_ratio: aspectRatio } : {}),
+        model: this.modelId,
+        ...(providerOptions.luma ?? {}),
+      },
+      abortSignal,
+      fetch: this.config.fetch,
+      failedResponseHandler: this.createLumaErrorHandler(),
+      successfulResponseHandler: createJsonResponseHandler(
+        lumaGenerationResponseSchema,
+      ),
+    });
+
+    const imageUrl = await this.pollForImageUrl(
+      generationResponse.id,
+      fullHeaders,
+      abortSignal,
+    );
+
+    const downloadedImage = await this.downloadImage(imageUrl, abortSignal);
+
+    return {
+      images: [downloadedImage],
+      warnings,
+      response: {
+        modelId: this.modelId,
+        timestamp: currentDate,
+        headers: responseHeaders,
+      },
+    };
+  }
+
+  private async pollForImageUrl(
+    generationId: string,
+    headers: Record<string, string | undefined>,
+    abortSignal: AbortSignal | undefined,
+  ): Promise<string> {
+    let attemptCount = 0;
+    const url = this.getLumaGenerationsUrl(generationId);
+    while (attemptCount < this.maxPollAttempts) {
+      const { value: statusResponse } = await getFromApi({
+        url,
+        headers,
+        abortSignal,
+        fetch: this.config.fetch,
+        failedResponseHandler: this.createLumaErrorHandler(),
+        successfulResponseHandler: createJsonResponseHandler(
+          lumaGenerationResponseSchema,
+        ),
+      });
+
+      switch (statusResponse.state) {
+        case 'completed':
+          if (!statusResponse.assets?.image) {
+            throw new InvalidResponseDataError({
+              data: statusResponse,
+              message: `Image generation completed but no image was found.`,
+            });
+          }
+          return statusResponse.assets.image;
+        case 'failed':
+          throw new InvalidResponseDataError({
+            data: statusResponse,
+            message: `Image generation failed.`,
+          });
+        case 'dreaming':
+        case 'queued':
+          break;
+      }
+
+      attemptCount++;
+      await new Promise(resolve =>
+        setTimeout(resolve, this.pollIntervalMillis),
+      );
```

Comment:
we should move `delay` from `ai` to `provider-utils` and use it here

Reply:
I will copy it for this PR and do the move in a second as it touches a batch of files in `ai`.